fed/mosaic-mapping/src/main/java/org/eclipse/mosaic/fed/mapping/ambassador/SpawningFramework.java

```diff
@@ -489,16 +489,14 @@ private void initTrafficLights(long time, RtiAmbassador rti, RandomNumberGenerat
 
             List<String> apps;
 
-            String name;
+            String name = UnitNameGenerator.nextTlName();
             String group;
             if (prototype != null) {
                 apps = prototype.getAppList();
                 group = ObjectUtils.defaultIfNull(prototype.getGroup(), tl.getGroupId());
-                name = UnitNameGenerator.nextTlName();
             } else {
                 apps = new ArrayList<>();
                 group = null;
-                name = tl.getGroupId();
```

Comment:
Was this a bug or intended behavior?

Reply:
I guess the upside of this was that there were no ID's handed out to TLs without applications. However, if we want the units to send interactions (i.e., the `TrafficLightSubscription`) they need a valid ID

Reply:
As an example if our SUMO scenario contains 5 TL-groups and we equip the first and the last group with an app, while disabling perception we would have logs for `tls_0` and `tls_4`. This may confuse users

Reply:
Assigning the groupId as the name of the unit looks very wrong to me. A traffic light unit with such a name would never be able to send or receive messages, as IpResolver can only resolve names with prefixes (e.g. `tl_`). Therefore I would accept your change without further investigation. 

However, just for clean code, can you rename `tl` to `tlGroup`?

Reply:
Yes exactly, I think that also was the intention behind this name-placeholder as traffic lights without names would never send or receive any messages. This also reduced the required names/addresses for traffic lights.